dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/AuthorityUtils.java

```diff
@@ -62,9 +64,33 @@ public String getPresentation(String schema, String element, String qualifier) {
      * @param projection the name of the projection to use, or {@code null}.
      * @return
      */
-    public AuthorityEntryRest convertEntry(Choice choice, String authorityName, Projection projection) {
-        AuthorityEntryRest entry = converter.toRest(choice, projection);
-        entry.setAuthorityName(authorityName);
+    public VocabularyEntryDetailsRest convertEntryDetails(Choice choice, String authorityName,
+           boolean isHierarchical, Projection projection) {
+        if (choice == null) {
+            return null;
+        }
+        VocabularyEntryDetailsRest entry = converter.toRest(choice, projection);
+        entry.setVocabularyName(authorityName);
+        entry.setId(authorityName + ":" + entry.getId());
+        entry.setInHierarchicalVocabulary(isHierarchical);
+        return entry;
+    }
+
+    public VocabularyEntryRest convertEntry(Choice choice, String authorityName, boolean storeAuthority,
```

Comment:
Please add JavaDocs to this new method

Reply:
https://github.com/DSpace/DSpace/pull/2792/files#diff-ce27772ea06f500b5b5d75b318f8369aR82-R92